exo-add/src/cli.ls

```diff
@@ -60,6 +62,15 @@ function help
   console.log help-message
 
 
+function check-for-service(app-config, data)
+  current-services = []
+  for protection-level of app-config.services
+    if app-config.services[protection-level]
+      current-services.push Object.keys(that)
+  if flatten current-services |> (.includes data.service-role )
+    console.log red "Service '#{data.service-role}' already exists in this application"
+    process.exit!
```

Comment:
Obey the law of demeter, and provide only the data that is needed by this method, now larger object that contain that data. And use prelude methods to solve this more elegantly:

```ls
function check-for-service {service-name, existing-services}
  existing-services |> (.values)
                    |> (.concat)
                    |> (.includes role)